### PR TITLE
Landing with aproach

### PR DESCRIPTION
**Pull request by hali9**
Implementation of the idea presented in:
https://github.com/iNavFlight/inav/issues/1814
https://github.com/iNavFlight/inav/issues/3670
and also
https://github.com/iNavFlight/inav/issues/4070
Tested and work well.
Need update configurator by add: "APROACH", "FS_ONLY_APR", "FS_NO_APR

### CODE REVIEW

File: src/main/navigation/navigation.h [hunk 1]
```diff
@@ -87,7 +87,10 @@ typedef enum {
 typedef enum {
     NAV_RTH_ALLOW_LANDING_NEVER = 0,
     NAV_RTH_ALLOW_LANDING_ALWAYS = 1,
-    NAV_RTH_ALLOW_LANDING_FS_ONLY = 2, // Allow landing only if RTH was triggered by failsafe
+    NAV_RTH_ALLOW_LANDING_APROACH = 2,
+    NAV_RTH_ALLOW_LANDING_FS_ONLY = 3,     // Allow landing only if RTH was triggered by failsafe
```
Comment (DzikuVx) [inline src/main/navigation/navigation.h]: @hali9 can we keep previously used IDs in place? So FS_ONLY stays at 2 and new modes go into new IDs?

**Review by DzikuVx — CHANGES_REQUESTED** [review]

File: src/main/navigation/navigation.h (hunk 1 above)
Comment (hali9) [reply src/main/navigation/navigation.h]: Yes, sure.

**Comment by MorrisR2** [comment]
What would need to be done to get this merged, other than fixing the merge conflicts?

**Comment by DzikuVx** [comment]
Replaced by #9713

**Pull request closed without merging**